src/pages/SelectClient/index.tsx

```diff
@@ -0,0 +1,190 @@
+  updateClient,
+  ClientId,
+} from '../../store/actions/clientActions';
+import { clientState } from '../../store/reducers/clientReducer';
+
+// Prop definitions
+interface OwnProps {}
+interface StateProps {
+  workflow: WorkflowStep;
+  chosenClients: clientState;
+}
+
+interface DispatchProps {
+  dispatchWorkflowUpdate: DispatchWorkflowUpdateType;
+  dispatchClientUpdate: DispatchClientUpdate;
+}
+type Props = StateProps & DispatchProps & OwnProps;
+
+const clientDetails = {
+  [ClientId.TEKU]: <TekuDetails shortened />,
+  [ClientId.LIGHTHOUSE]: <LighthouseDetails shortened />,
+  [ClientId.PRYSM]: <PrysmDetails shortened />,
+  [ClientId.NIMBUS]: <NimbusDetails shortened />,
+  [ClientId.GETH]: <GethDetails />,
+  [ClientId.OPEN_ETHEREUM]: <OpenEthereumDetails />,
+  [ClientId.BESU]: <BesuDetails />,
+  [ClientId.NETHERMIND]: <NethermindDetails />,
+};
+
+export type Client = {
+  clientId: ClientId;
+  name: string;
+  imgUrl: string;
+};
+
+// define and shuffle the clients
+const ethClients: {
+  [ethVersion: number]: Array<Client>;
+} = {
+  1: _shuffle([
+    {
+      clientId: ClientId.OPEN_ETHEREUM,
+      name: 'OpenEthereum',
+      imgUrl: OpenEthereumCircle,
+    },
+    { clientId: ClientId.GETH, name: 'Geth', imgUrl: GethCircle },
+    { clientId: ClientId.BESU, name: 'Besu', imgUrl: BesuCircle },
+    {
+      clientId: ClientId.NETHERMIND,
+      name: 'Nethermind',
+      imgUrl: NethermindCircle,
+    },
+  ]),
+  2: _shuffle([
+    { clientId: ClientId.TEKU, name: 'Teku', imgUrl: TekuCircle },
+    {
+      clientId: ClientId.LIGHTHOUSE,
+      name: 'Lighthouse',
+      imgUrl: LighthouseCircle,
+    },
+    {
+      clientId: ClientId.PRYSM,
+      name: 'Prysm',
+      imgUrl: PrysmaticCircle,
+    },
+    { clientId: ClientId.NIMBUS, name: 'Nimbus', imgUrl: NimbusCircle },
+  ]),
+};
+
+const _SelectClientPage = ({
+  workflow,
+  dispatchWorkflowUpdate,
+  chosenClients,
+  dispatchClientUpdate,
+}: Props): JSX.Element => {
+  // set the default the eth version to 1 on initial render
+  const [ethVersionStep, setEthVersionStep] = useState<1 | 2>(1);
+
+  // filter the options based on the eth version the user is on
+  const clientOptions = React.useMemo(() => ethClients[ethVersionStep], [
+    ethVersionStep,
+  ]);
+
+  // memoize the chosen client by step
+  const selectedClient: ClientId = React.useMemo(
+    () =>
+      ethVersionStep === 1
+        ? chosenClients.eth1Client
+        : chosenClients.eth2Client,
+    [ethVersionStep, chosenClients]
+  );
+
+  const setClientFxn = (clientId: ClientId) => {
+    dispatchClientUpdate(clientId, ethVersionStep);
+  };
+
+  React.useEffect(() => {
+    const header = document.getElementsByTagName('header')[0];
+
+    if (header) {
+      header.scrollIntoView({ behavior: 'smooth' });
+    }
+  }, [ethVersionStep]);
+
+  const handleSubmit = () => {
+    if (workflow === WorkflowStep.SELECT_CLIENT) {
+      dispatchWorkflowUpdate(WorkflowStep.TRANSACTION_SIGNING);
```

Comment:
should it be `GENERATE_KEY_PAIRS`?
```suggestion
      dispatchWorkflowUpdate(WorkflowStep.GENERATE_KEY_PAIRS);
```

Reply:
I think you are correct here, but I cannot get the bad behavior to occur.

Reply:
I found it since the checkbox on generate-keys page was always *checked* by default when I first visit this page. (which should be check by users themselves)

![image](https://user-images.githubusercontent.com/9263930/94002846-0a3b4180-fdcd-11ea-89f3-287a31433745.png)


I just tried changed it to `dispatchWorkflowUpdate(WorkflowStep.GENERATE_KEY_PAIRS);` locally. The checkbox became unchecked by default. :)